storage/test/sparse_merkle/internal_node_test.cpp

```diff
@@ -480,6 +485,64 @@ TEST(remove_tests, remove_a_leaf_with_a_peer) {
   ASSERT_EQ(child2, promoted);
 }
 
+/*
+// Add 2 LeafChildren to a BatchedInternalNode and then remove one of them.
+//
+// The logical tree inside the BatchedInternalNode looks like the following
+// before remove is called:
+//
+//          Root
+//           |
+//         /   \
+//      Leaf1   Leaf2
+//
+// Since this is the root BatchedInternalNode the peer will not be promoted.
+// The logical tree will look like the following after the remove is called:
+//
+//            Root
+//             |
+//     -----------------
+//     |               |
+// Placeholder       Leaf2
+//
+*/
+TEST(remove_tests, remove_a_leaf_with_a_peer_root_node) {
+  BatchedInternalNode node;
+
+  Hasher hasher;
+  const char* key1 = "artist";
+  const char* value1 = "REM";
+  const char* value2 = "Nas";
+  size_t depth = 0;
+
+  auto key1_hash = hasher.hash(key1, strlen(key1));
+
+  // Flip the first bit so that key1_hash becomes a sibling of key2_hash
+  auto key2_hash = flipByte0Bit(0, key1_hash);
+
+  ASSERT_EQ(0, key1_hash.prefix_bits_in_common(key2_hash));
+
+  auto value1_hash = hasher.hash(value1, strlen(value1));
+  auto value2_hash = hasher.hash(value2, strlen(value2));
+  auto leaf_key1 = LeafKey(key1_hash, Version(1));
+  auto leaf_key2 = LeafKey(key2_hash, Version(2));
+  auto child1 = LeafChild{value1_hash, leaf_key1};
+  auto child2 = LeafChild{value2_hash, leaf_key2};
+
+  node.insert(child1, depth);
+  node.insert(child2, depth);
+
+  // There should be one root and two leaves
+  ASSERT_EQ(3, node.numChildren());
+  ASSERT_EQ(1, node.numInternalChildren());
+  ASSERT_EQ(2, node.numLeafChildren());
+
+  // Deleting key1 should return RemoveComplete, since the peer cannot be promoted above the root node.
+  auto result = node.remove(key1_hash, depth, Version(3));
+  ASSERT_TRUE(std::holds_alternative<BatchedInternalNode::RemoveComplete>(result));
+  ASSERT_EQ(node.hash(), hasher.parent(PLACEHOLDER_HASH, value2_hash));
+}
```

Comment:
Maybe we can verify the children count after the `remove` call, something like:
```
  // There should be one root and one leaf
  ASSERT_EQ(2, node.numChildren());
  ASSERT_EQ(1, node.numInternalChildren());
  ASSERT_EQ(1, node.numLeafChildren());
```

Reply:
Great suggestion. Also, thank you for taking the time to understand these tests :)

Reply:
Fixed.